### PR TITLE
Add frequent partners to config file for Defender 2.3

### PR DESCRIPTION
**Pull request by Sloane4**
#  #

## 🗣 Description ##

Assert that frequent partner domains are added to impersonation protection in both the standard and strict preset policies.
Closing #490 

## 💭 Motivation and context ##

Defender 2.3 specifies that "frequent partners" should be added to the strict preset EOP policy. By allowing the users to specify these in the config file, we can verify the correct domains are added.

## 🧪 Testing ##

-  Unit Testing

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

File: sample-config-files/defender-config.yaml [hunk 1]
```diff
@@ -46,4 +46,8 @@ Defender:
   MS.DEFENDER.2.1v1 : &UserImpersonationProtection
     SensitiveAccounts:
       IncludedUsers:
-        - "John Doe;jdoe@someemail.com"
\ No newline at end of file
+        - "John Doe;jdoe@someemail.com"
+  MS.DEFENDER.2.3v1 : &DomainImpersonationProtection
+    SensitiveAccounts:
+      IncludedDomains:
+        - "random.mail.example.com"
```
Comment (adhilto) [inline sample-config-files/defender-config.yaml:53]: Nitpick. Just wondering about consistency. Some of these are wrapped in quotes, e.g., `"random.mail.example.com"`, but not `johndoe@random.example.com`. Both styles are valid. But maybe we'd want to have all with quotes or all without for consistency.

**Review by adhilto — APPROVED** [review]
Looks great. Just one nitpick comment, but either way, approving.

File: sample-config-files/defender-config.yaml (hunk 1 above)
Comment (Sloane4) [reply sample-config-files/defender-config.yaml]: Copy and paste error, remove the string quotes

File: Rego/Utils/DefenderUtils.rego [hunk 2]
```diff
@@ -125,40 +125,45 @@ SensitiveAccounts(SensitiveAccountsSetting, SensitiveAccountsConfig) := true if
...
+        "Action": Policy[ActionKey]
     }
 } if {
-    Policy := ImpersonationProtectionSetting(Policies, IdentityString)
-    count(Policy[0]) > 0
+    Policy := ImpersonationProtectionSetting(Policies, IdentityString, FilterKey)
+    count(Policy) > 0
+
+    PolicyProtectedAccounts := { lower(x) | some x in Policy[AccountKey] }
+    print(IncludedAccounts - PolicyProtectedAccounts)
```
[the remark this answers is not in the document]
Comment (Sloane4) [reply Rego/Utils/DefenderUtils.rego]: removed please confirm

**Comment by Sloane4** [comment]
@nanda-katikaneni ready to merge

**Pull request merged**